streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorContextUtils.java

```diff
@@ -56,18 +56,26 @@ public static StreamsMetricsImpl getMetricsImpl(final StateStoreContext context)
         return (StreamsMetricsImpl) context.metrics();
     }
 
-    public static String changelogFor(final ProcessorContext context, final String storeName) {
+    public static String changelogFor(final ProcessorContext context, final String storeName, final Boolean newChangelogTopic) {
         final String prefix = getPrefix(context.appConfigs(), context.applicationId());
-        return context instanceof InternalProcessorContext
-            ? ((InternalProcessorContext) context).changelogFor(storeName)
-            : ProcessorStateManager.storeChangelogTopic(prefix, storeName, context.taskId().topologyName());
+        if (context instanceof InternalProcessorContext && !newChangelogTopic) {
+            final String changelogTopic = ((InternalProcessorContext) context).changelogFor(storeName);
+            if (changelogTopic != null)
+                return changelogTopic;
+
+        }
+        return ProcessorStateManager.storeChangelogTopic(prefix, storeName, context.taskId().topologyName());
     }
 
-    public static String changelogFor(final StateStoreContext context, final String storeName) {
+    public static String changelogFor(final StateStoreContext context, final String storeName, final Boolean newChangelogTopic) {
         final String prefix = getPrefix(context.appConfigs(), context.applicationId());
-        return context instanceof InternalProcessorContext
-            ? ((InternalProcessorContext) context).changelogFor(storeName)
-            : ProcessorStateManager.storeChangelogTopic(prefix, storeName, context.taskId().topologyName());
+        if (context instanceof InternalProcessorContext && !newChangelogTopic) {
+            final String changelogTopic = ((InternalProcessorContext) context).changelogFor(storeName);
```

Comment:
I checked the code in `InternalTopologyBuilder` and I think that at the time `ProcessorStateManager` is constructed all store->changelog pairs should be inside the `Map<String, String> storeToChangelogTopic` already, and assuming that for all mock/tests, we let the mock of InternalProcessorContext to return non-null strings we should never have `null` returned from `changelogFor`. @cadonna could you help me double check here, since in the original PR your intention (https://github.com/apache/kafka/pull/8902/files#r452148874) is to still need null-check.

If that's the case, we can further simply the logic here as:

1) we would never need to fallback to `ProcessorStateManager.storeChangelogTopic` since the changelog string should be constructed via the function and added into the map in `InternalTopologyBuilder` as always.
2) hence we do not need the third param since we would never need to create new changelog.